core-ui/src/components/Extensibility/components/Badge.js

```diff
@@ -1,16 +1,39 @@
 import React from 'react';
 import { isNil } from 'lodash';
+import jsonata from 'jsonata';
 
 import { StatusBadge } from 'shared/components/StatusBadge/StatusBadge';
 import { useGetPlaceholder } from 'components/Extensibility/helpers';
 
-export function Badge({ value, structure, schema }) {
+export function Badge({ value, structure, schema, ...props }) {
   const { emptyLeafPlaceholder } = useGetPlaceholder(structure);
+
+  let type = null;
+  if (structure.highlights) {
+    const match = Object.entries(structure.highlights).find(([key, rule]) => {
+      if (Array.isArray(rule)) {
+        return rule.includes(value);
+      } else {
+        try {
+          return jsonata(rule).evaluate({ value });
```

Comment:
We use `data` while passing stuff to jsonata, let's be consistent.
```suggestion
          return jsonata(rule).evaluate({ data: value });
```